### ISSUE TITLE
SPIKE - metadata history from DataHub

### ISSUE THREAD

**Issue by jemnery** [ministryofjustice/find-moj-data#129]

**Comment by tom-webber** [comment]
[Relevant comment in DataHub slack](https://datahubspace.slack.com/archives/CV2KB471C/p1701970434619739?thread_ts=1701679783.644839&cid=CV2KB471C)

**Comment by MatMoore** [comment]
Previously:

https://github.com/ministryofjustice/data-platform/issues/3153
https://github.com/ministryofjustice/data-platform/issues/3402

## If we use datahub's API to pull the top level figures
Then the answer is **no**

The data comes from getHighlights, which takes no arguments, so we would have to capture data over time 
https://datahubproject.io/docs/graphql/queries#gethighlights

## If we query datahub's DB directly
Then the answer is TBC

**Comment by jemnery** [comment]
Agree, not a good idea to hit the DB directly. We're not protected from schema changes, and we could end up impacting performance. I'd assume the DH DB was tuned with indexes etc to optimise API calls, and us hitting it with ad-hoc queries could result in slower, resource hungry table scans 🤷 

In other words, a separated rolling "cache" of metadata history will mean running metadata dashboards doesn't impact on the app's performance.

The only downside is if / when we want to introduce new metrics at a later date these won't be cached. In that case running a one-off ad-hoc query to fetch the historical data is probably acceptable.
